lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -192,6 +192,113 @@ LocaleMatch lookupMatcher(
+  // 6. If type is "string", then
+  // a. Set value to ? ToString(value).
+  // 7. If values is not undefined and values does not contain an element equal
+  // to value, throw a RangeError exception.
+  if (!value->second.isString() ||
+      (!values.empty() &&
+       llvh::find(values, value->second.getString()) == values.end())) {
+    return runtime->raiseRangeError(
+        vm::TwineChar16("Value ") +
+        vm::TwineChar16(value->second.getString().c_str()) +
+        vm::TwineChar16(
+            " out of range for Intl.DateTimeFormat options property ") +
+            vm::TwineChar16(property.c_str()));
+  }
+  // 8. Return value.
+  return llvh::Optional<std::u16string>(value->second.getString());
+}
+vm::CallResult<llvh::Optional<bool>> getOptionBool(
+    const Options &options,
+    const std::u16string &property,
+    llvh::Optional<bool> fallback,
+    vm::Runtime *runtime) {
+  //  1. Assert: Type(options) is Object.
+  //  2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  //  3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return fallback;
+  }
+  if (!value->second.isBool()) {
+    return runtime->raiseRangeError(
+        vm::TwineChar16("Value ") +
+        vm::TwineChar16(value->second.getBool()) +
+        vm::TwineChar16(
+            " out of range for Intl.DateTimeFormat options property ") +
+            vm::TwineChar16(property.c_str()));
+  }
+  //  8. Return value.
+  return llvh::Optional<bool>(value->second.getBool());
+}
+// https://tc39.es/ecma402/#sec-defaultnumberoption
+vm::CallResult<llvh::Optional<uint8_t>> defaultNumberOption(
+    const double value,
+    const std::uint8_t minimum,
+    const std::uint8_t maximum,
+    llvh::Optional<uint8_t> fallback,
+    vm::Runtime *runtime) {
+  //  1. If value is undefined, return fallback.
+  if (!value && value != 0) {
+    return fallback;
+  }
+  //  2. Set value to ? ToNumber(value).
+  //  3. If value is NaN or less than minimum or greater than maximum, throw a
+  //  RangeError exception.
+  if (std::isnan(value) || value < minimum || value > maximum) {
+    return vm::ExecutionStatus::EXCEPTION;
+  }
+  //  4. Return floor(value).
+  return llvh::Optional<uint8_t>(std::floor(value));
+}
+// Implementation of
+// https://402.ecma-international.org/8.0/#sec-getnumberoption
+vm::CallResult<llvh::Optional<uint8_t>> getNumberOption(
+    const Options &options,
+    const std::u16string &property,
+    const std::uint8_t minimum,
+    const std::uint8_t maximum,
+    llvh::Optional<uint8_t> fallback,
+    vm::Runtime *runtime) {
+  //  1. Assert: Type(options) is Object.
+  //  2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  if (value == options.end()) {
+    return fallback;
+  }
+  //  3. Return ? DefaultNumberOption(value, minimum, maximum, fallback).
+  auto defaultNumber = defaultNumberOption(
+      value->second.getNumber(),
```

Comment:
We don't know that this property is a number, since we haven't checked it. The check will be performed in `defaultNumberOption`.